src/gleam/http.gleam

```diff
@@ -556,3 +556,28 @@ pub fn expire_resp_cookie(resp, name, attributes) {
   let attrs = CookieAttributes(..attributes, max_age: Some(0))
   set_resp_cookie(resp, name, "", attrs)
 }
+
+/// Get the origin request header
+///
+/// If no "Origin" header is found in the request, falls back to the "Referer"
+/// header.
+pub fn get_req_origin(req: Request(body)) -> Option(String) {
+  case get_req_header(req, "origin") {
+    Ok(origin) -> Some(origin)
+    Error(Nil) ->
+      case get_req_header(req, "referer") {
+        Ok(ref) ->
+          case ref
+          |> uri.parse {
+            Ok(uri) ->
```

Comment:
There is a uri.origin function https://hexdocs.pm/gleam_stdlib/gleam/uri/#origin

Reply:
Updated to use the `uri.origin` function